pkg/coordinatormanager/coordinator_informer.go

```diff
@@ -235,19 +240,28 @@ func (cc *CoordinatorController) addEventHandlers(
 		return err
 	}
 
+	// check the current k8s whether registered the ServiceCIDR resource
 	InformerLogger.Debug("Checking if the ServiceCIDR is available in your cluster")
-	var serviceCIDR networkingv1alpha1.ServiceCIDRList
-	err = cc.APIReader.List(context.TODO(), &serviceCIDR)
-	if err != nil {
-		InformerLogger.Warn("ServiceCIDR feature is unavailable in your cluster, Don't start the serviceCIDR informer")
-		return nil
+	resourceList, err := cc.K8sClient.DiscoveryClient.ServerResourcesForGroupVersion(networkingv1alpha1.SchemeGroupVersion.String())
+	if nil != err {
+		return fmt.Errorf("no '%s' API Version resources found in the current kubernetes cluster, error: %v", networkingv1alpha1.SchemeGroupVersion.String(), err)
+	}
+	for _, apiResources := range resourceList.APIResources {
+		if apiResources.Kind == constant.KindServiceCIDR {
+			cc.ServiceCIDR.IsAvailable = true
+			break
+		}
 	}
 
-	InformerLogger.Debug("the ServiceCIDR is available in your cluster, Start the serviceCIDR informer")
-	if err = cc.addServiceCIDRHandler(serviceCIDRInformer.Informer()); err != nil {
-		return err
+	if cc.ServiceCIDR.IsAvailable {
+		InformerLogger.Debug("the ServiceCIDR is available in your cluster, Start the serviceCIDR informer")
+		if err = cc.addServiceCIDRHandler(serviceCIDRInformer.Informer()); err != nil {
+			return err
+		}
+	} else {
+		InformerLogger.Warn("ServiceCIDR feature is unavailable in your cluster, Don't start the serviceCIDR informer")
 	}
```

Comment:
Just use the kubernetes API resource to check whether the ServiceCIDR resource exists in the current cluster.
Reference previous PR: https://github.com/spidernet-io/spiderpool/pull/3185